docs/design/datamover.md

```diff
@@ -63,93 +63,85 @@ This design supports adding the data mover feature to the OADP operator and faci
+### Volume Snapshot Backup
 
+Assuming that the `DataMover Enable` flag is set to true in the DPA config, when a velero backup is created, it triggers the custom velero CSI plugin plugin (velero BackupItemAction plugin) to create the `VolumeSnapshotBackup` CR in the app namespace. The extended plugin looks up for the PVCs in the user namespace mentioned in the velero backup and creates a `VolumeSnapshotBackup` CR for every PVC in that namespace.
 
-```
-apiVersion: oadp.openshift.io/v1alpha1
-kind: DataMoverClass
-metadata:
-  annotations:
-    oadp.openshift.io/default: "true"
-  name: <name>
-spec:
-  mover: <VolSync>
-  selector: <tagname>
-
-```
-
-The above `DataMoverClass` name will be referenced in `DataMoverBackup` & `DataMoverRestore` CRs. This will help in selecting the data mover implementation during runtime. If the `DataMoverClass` name is not defined, then the default `DataMoverClass` will be used, which in this case will be `VolSync`
-
-### Data Mover Backup
-
-Assuming that the `DataMover Enable` flag is set to true in the DPA config, when a velero backup is created, it triggers the custom velero CSI plugin plugin (velero BackupItemAction plugin) to create the `DataMoverBackup` CR in the app namespace. The extended plugin looks up for the PVCs in the user namespace mentioned in the velero backup and creates a `DataMoverBackup` CR for every PVC in that namespace that is filtered by the `datamoverclass.spec.selector`.
-
-`DataMoverBackup` CR supports either a volumesnapshot or a pvc as the type of the backup object. If the velero CSI plugin is used for backup, `VolumeSnapshot` is used as the type or else `PVC`
-is used.
+`VolumeSnapshotBackup` CR supports a `volumesnapshotcontent` as the type of the backup object. Velero backup will wait for the `VolumeSnapshotBackup` to be complete. Once all the `VolumeSnapshotBackup` gets completed, Velero backup's status get updated accordingly.
 
 ```
-apiVersion: oadp.openshift.io/v1alpha1
-kind: DataMoverBackup
+apiVersion: datamover.oadp.openshift.io/v1alpha1
+kind: VolumeSnapshotBackup
 metadata:
-  name: <name>
-spec:
-  dataMoverClass: <DataMoverClass name> 
-  dataSourceRef:
-    apiGroup: <APIGroup>
-    kind: <PVC|VolumeSnapshotContent>
-    name: <name>
-  config:  //optional based on the datamover impl
-
-```
-### Data Mover Restore
-When a velero restore is triggered, the custom Velero CSI plugin looks for `DataMoverBackup` in the backup resources. If it encounters a `DataMoverBackup` resource, then the extended plugin (velero RestoreItemAction plugin) will create a `DataMoverRestore` CR in the app namespace. It will populate the CR with the details obtained from the `DataMoverBackup` resource. 
-
-```
-apiVersion: oadp.openshift.io/v1alpha1
-kind: DataMoverRestore
-metadata:
-  name: <name>
+  annotations:
+    datamover.io/restic-repository: <restic_repo>
+    datamover.io/source-pvc-name: <src_pvc_name>
+    datamover.io/source-pvc-size: <src_pvc_size>
```

Comment:
Latest VSB has annotations regarding storageclass name and volumesnapshotclass name as well.